shared/actions/chat2/index.js

```diff
@@ -565,169 +564,153 @@ const reactionUpdateToActions = (info: RPCChatTypes.ReactionUpdateNotif) => {
   return [Chat2Gen.createUpdateReactions({conversationIDKey, updates})]
 }
 
+// This is to simplify the changes that setIncomingCallMap created. Could clean this up and remove this
+const arrayOfActionsToSequentially = actions =>
+  Saga.call(Saga.sequentially, (actions || []).map(a => Saga.put(a)))
+
 // Handle calls that come from the service
 const setupEngineListeners = () => {
-  engine().setIncomingActionCreators(
```

Comment:
this is the big change. instead of 
`setIncomingActionCreators('method', () => action) // n times`
its now
`setIncomingCallMap({method: () => effect})`
which matches exactly what you do with the engine saga helpers